src/mame/namco/namcos10.cpp

```diff
@@ -748,6 +955,18 @@ void namcos10_state::namcos10_base(machine_config &config)
 
 void namcos10_state::machine_start()
 {
+	m_psx_remap_timer = timer_alloc(FUNC(namcos10_state::psx_remap_tick), this);
+	m_psx_remap_timer->adjust(attotime::never);
+
+	m_maincpu->space(AS_PROGRAM).install_write_tap(
+		0x1f801010, 0x1f801013,
+		"rom_configure_tap_w",
+		[this] (offs_t offset, u32 &data, u32 mem_mask) {
+			// reinstall memory map after rom_config_w finishes with update_rom_config
+			m_psx_remap_timer->adjust(attotime::zero);
+		}
+	);
```

Comment:
This feels like a gross way of doing this, relying on too much knowledge of how the PSX CPU device works.

It would be best if the CPU could just get a bus error on unmapped accesses so it didn’t need to nuke everything in the address space – @galibert can the memory system do this yet, or do we still need to rely on installing bus error handlers?

The CPU could expose a notification when the when it reconfigures its mappings (see src/lib/util/notifier.h – it’s used in src/emu/emumem.h for address space mapping change notifications, and for various things by the Lua engine in src/frontend/mame/luaengine.cpp).

If you want to do it non-intrusively, you can add a change notifier on the address space (see `add_change_notifier`) and use a guard variable to ensure you don’t keep scheduling yourself reentrantly.

And that aside, you don’t need an allocated timer for this, you can just use `machine().scheduler().synchronize()` to get yourself a callback from the scheduler at the next end-of-timeslice.

@galibert do you have any more thoughts on the best way to deal with the PSX CPU nuking the input mappings?

Reply:
This code was more of a compromise because it was a hard blocker for the rest of the driver but I also didn't want to touch the PSX core code. `add_change_notifier` looks interesting though, I'll read that code a little more to see if I can apply it here.

Reply:
I tried `add_change_notifier` but it doesn't trigger anything at all when I use it on the program space of m_maincpu. Just to test I even tried it on the `m_program` stored inside the PSX core itself and even that doesn't trigger anything.

Reply:
You did hold on to the subscription returned by `add_change_notifier`, I hope?  If you don’t store it somewhere, it will be immediately destructed, and you’ll be unsubscribed immediately (hence receiving no notifications of changes).

Reply:
Yeah... that was the problem. Didn't realize. It's working now and looks much cleaner than the other things. I tried it with Gahaha (which needs analog I/O so it's an easy test) and also a game that doesn't use a remapper (Medal no Tatsujin, I/O board not emulated yet so no mapping needed) and it seems to be working fine. Thanks for being patient, didn't even consider it would unsubscribe if I didn't store it.